### PR TITLE
fix(main): fix ImportError on examples dags

### LINKED ISSUE #945: Quickstart: not having `virtualenv` installed results in broken DAGs
Error-free installation of Airflow using the [Quickstart](https://airflow.apache.org/docs/apache-airflow/stable/start.html) seems to require `virtualenv`, but there is no mention of this in the guide.

The following example DAGs have the dependency:
- example_branch_operator_decorator
- example_branch_operator

Screenshot 2024-01-29 at 10 55 06 PM

### PR DESCRIPTION
**Pull request by xionams**
this pull request is related to: https://github.com/apache/airflow-site/issues/945
closes: https://github.com/apache/airflow-site/issues/945

the solution of "skipped" on the dag-loading when the virtualenv is not installed
is wrap the dag loading with try-except and return None either
using this logic will guarantee the ImportError exception will not stored in the DB 
and not thrown on the homepage as an error.
**so if virtualenv is not installed the examples are not loaded and the exception will not thrown**

tested on Ubuntu WSL2 

---

### CODE REVIEW

File: airflow/example_dags/libs/helper.py [hunk 1]
```diff
@@ -20,3 +20,11 @@
...
 def print_stuff():
     print("annotated!")
+
+def check_venv_dependency():
+    """Check if the virtualenv module is installed"""
+    try:
+        import virtualenv
+        return True
+    except ImportError:
+        return False
```
Comment (eladkal) [inline airflow/example_dags/libs/helper.py]: You can use
`from airflow.operators.python import is_venv_installed`

https://github.com/apache/airflow/blob/8c0a4de66344647c759b6188cabafacf26ace9e3/airflow/operators/python.py#L66-L74

File: airflow/example_dags/example_branch_operator.py [hunk 2]
```diff
@@ -28,140 +28,147 @@
...
-    tags=["example", "example2"],
-    orientation="TB",
-) as dag:
-    run_this_first = EmptyOperator(
-        task_id="run_this_first",
+from airflow.operators.python import is_venv_installed
+
+venv_installed=is_venv_installed()
+
+if venv_installed:
```
Comment (Taragolis) [inline airflow/example_dags/example_branch_operator.py]:
```suggestion
if is_venv_installed():
```

File: airflow/example_dags/example_branch_operator.py [hunk 3]
```diff
@@ -28,140 +28,147 @@
...
+                task_id=f"venv_{option}",
+                requirements=["numpy~=1.24.4"],
+                venv_cache_path=VENV_CACHE_PATH,
+                python_callable=hello_world_with_venv,
+            )
+
+            # Label is optional here, but it can help identify more complex branches
+            branching_venv >> Label(option) >> t >> join_venv
+else:
+    dag = None
```
Comment (Taragolis) [inline airflow/example_dags/example_branch_operator.py]:
```suggestion
```

File: airflow/example_dags/example_branch_operator_decorator.py [hunk 4]
```diff
@@ -29,116 +29,124 @@
...
-with DAG(
-    dag_id="example_branch_python_operator_decorator",
-    start_date=pendulum.datetime(2021, 1, 1, tz="UTC"),
-    catchup=False,
-    schedule="@daily",
-    tags=["example", "example2"],
-    orientation="TB",
-) as dag:
-    run_this_first = EmptyOperator(task_id="run_this_first")
+if venv_installed:
```
Comment (Taragolis) [inline airflow/example_dags/example_branch_operator_decorator.py]:
```suggestion
if is_venv_installed():
```

File: airflow/example_dags/example_branch_operator_decorator.py [hunk 5]
```diff
@@ -29,116 +29,124 @@
...
 
-from airflow.decorators import task
-from airflow.models.dag import DAG
-from airflow.operators.empty import EmptyOperator
-from airflow.utils.edgemodifier import Label
-from airflow.utils.trigger_rule import TriggerRule
+from airflow.operators.python import is_venv_installed
 
-PATH_TO_PYTHON_BINARY = sys.executable
+venv_installed=is_venv_installed()
```
Comment (Taragolis) [inline airflow/example_dags/example_branch_operator_decorator.py]:
```suggestion
```

File: airflow/example_dags/example_branch_operator_decorator.py [hunk 6]
```diff
@@ -29,116 +29,124 @@
...
-        t = some_venv_task()
+            t = some_venv_task()
 
-        # Label is optional here, but it can help identify more complex branches
-        random_choice_venv >> Label(option) >> t >> join_venv
+            # Label is optional here, but it can help identify more complex branches
+            random_choice_venv >> Label(option) >> t >> join_venv
+else:
+    deg = None
+
```
Comment (Taragolis) [inline airflow/example_dags/example_branch_operator_decorator.py]:
```suggestion
```

**Comment by potiuk** [comment]
Just one static check to fix

File: tests/system/README.md [hunk 7]
```diff
@@ -22,6 +22,7 @@
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
+
```
Comment (eladkal) [inline tests/system/README.md]:
```suggestion
```

File: dev/PROJECT_GUIDELINES.md [hunk 8]
```diff
@@ -20,6 +20,7 @@
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
+
```
Comment (eladkal) [inline dev/PROJECT_GUIDELINES.md]:
```suggestion
```

**Comment by eladkal** [comment]
i submitted fixes for the static checks. It should pass now

**Review by potiuk — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**